app/api/routes/product.py

```diff
@@ -0,0 +1,152 @@
+    try:
+        product_uuid = UUID(product_id)
+        product = products_db.get(product_id)
+        if not product:
+            raise HTTPException(status_code=404, detail=f"Product with ID {product_id} not found.")
+    except ValueError:
+        raise HTTPException(status_code=400,detail="Product ID must be a valid integer." )
+    
+    return product
+
+"""Update products by the ID"""
+@router.put("/products/{product_id}", status_code=200 ,response_model=Product)
+async def update_product(product_id: UUID,update_data: ProductUpdate,admin: bool = Depends(get_current_admin)):
+    
+    try:
+        product = products_db.get(product_id)
+        raise HTTPException(status_code=404,detail=f"Product with ID {product_id} not found.")
+        if update_data.name:
+            if any(existing_product['name'].lower() == update_data.name.lower() for existing_product in products_db.values()):
+                raise HTTPException(status_code=400,detail=f"Product name '{update_data.name}' already exists. Please use a unique name.")
+        product.name = update_data.name
+        if update_data.price is not None:
+            if update_data.price < 0:
+                raise HTTPException(status_code=400,detail="Price must be a non-negative number.")
+        product.price = update_data.price
+        if update_data.stock is not None:
+            if update_data.stock < 0:
+                raise HTTPException(status_code=400,detail="Stock must be a non-negative number.")
+        product.stock = update_data.stock
+    except Exception as e:
+        raise (e) 
+    
+
+"""Delete product by the ID"""
+@router.delete("/products/{product_id}", status_code=200)
+async def delete_product(product_id: UUID):
+    try:
+        if product_id not in products_db:
+            raise HTTPException(status_code=404, detail=f"Product with ID {product_id} not found.")
+        del products_db[product_id]
+        return {"message": "Product deleted successfully"}
+    except KeyError:
+        raise HTTPException(status_code=500, detail="An error occurred while deleting the product.")
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+""" List Products"""
+@router.get("/products", status_code=200)
+async def list_products():
+    if not products_db:
+        return []
+    
+    products_list = []
+    for product_id, product_data in products_db.items():
+        product_dict = {
+            "id": product_id,
+            "name": product_data["name"],
+            "price": product_data["price"],
+            "description": product_data.get("description", ""), 
+            "stock": product_data.get("stock", 0),  
+            "isAvailable": product_data.get("isAvailable", True),  
+            "created_at": product_data["created_at"]
+        }
+        products_list.append(product_dict)
+    return products_list
+    
+""" Search for Product"""
+@router.get("/products/search",status_code=200,response_model=dict)
```

Comment:
This logic will be changed when connecting DB, i will ignore it for now

Reply:
Forsure